src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -864,6 +864,23 @@ private static bool PackBlocks(ArrayBuilder<BasicBlockBuilder> blocks, PooledDic
                                         continue;
                                     }
 
+                                    if (currentRegion.LastBlock == block &&
+                                        block.BranchValue is IFlowCaptureReferenceOperation { Id: var id } &&
+                                        currentRegion.HasCaptureIds &&
+                                        currentRegion.CaptureIds.Contains(id) &&
```

Comment:
>currentRegion.CaptureIds.Contains(id) [](http://example.com/codeflow?start=40&length=37)

It is quite possible that with switch expressions the result capture is necessarily owned by the immediate enclosing reagion. However, I think in general the capture could be from parent region and the block is the last block in it. Even though we don't have a test scenario for that readily available, consider expanding the check to captures in the parent regions as well, as long as the block is the last block in them.